hier_config/__init__.py

```diff
@@ -61,6 +61,15 @@ def __init__(self, hostname, os, options):
         self.children = []
         self.children_dict = {}
 
+        if self.hostname and self.os and self.options:
```

Comment:
- Use `all` to ensure `hostname`, `os`, and `options` `is not None`
- `elif host is not None`
- Add asserts to ensure `Host` `hasattrs` `os` and `hostname`